o1vm/src/interpreters/mips/witness.rs

```diff
@@ -314,17 +329,17 @@ impl<Fp: Field, PreImageOracle: PreImageOracleT> InterpreterEnv for Env<Fp, PreI
 
     fn is_zero(&mut self, x: &Self::Variable) -> Self::Variable {
         // write the result
-        let pos = self.alloc_scratch();
-        let res = if *x == 0 { 1 } else { 0 };
-        self.write_column(pos, res);
+        let res = {
+            let pos = self.alloc_scratch();
+            unsafe { self.test_zero(x, pos) }
+        };
         // write the non deterministic advice inv_or_zero
-        let pos = self.alloc_scratch();
-        let inv_or_zero = if *x == 0 {
-            Fp::zero()
+        let pos = self.alloc_scratch_inverse();
+        if *x == 0 {
```

Comment:
I am not sure why there's this `if-else` distinction if it always writes in the inverse scratch state. Is it to avoid a conversion in case `Fp::zero()` is faster than `Fp:from(0)`? Either way I suppose the batch inversion algorithm will just ignore zeros, I believe I checked that in the arkworks code.

Reply:
No reason. Seems to be an ugly code :sweat_smile:. For a follow-up.